### PR TITLE
Fix for language selection error when updating

### LINKED ISSUE #20960: Language selection in Updater no longer works
## Context
When a database update is required Matomo displays an update screen. By default the screen uses the browser language.

There is also a language drop down, which should allow to change the language of the screen.

image

Selecting another language, currently tries to save the selected language by posting a form to `index.php?module=LanguagesManager&amp;action=saveLanguage`

As the updater might be performed without a session or login, this can result in an error screen, asking to login.

image

## Expected Behavior

The language of the update screen can simply be changed by using the parameter `&language=XX`, so when choosing another language the parameter in url should be change/added.

## Steps to Reproduce (for Bugs)
1. Force update screen to display by changing the option value for `version_Core` to a value like `4.14.0`
2. Open Matomo
3. Try to select another language

## Your Environment

* Matomo Version: 5.0.0-b4
* PHP Version: 7.2
* Server Operating System: Linux

### PR DESCRIPTION
**Pull request by bx80**
### Description:

Fixes #20960 

This PR disables the CSRF check for the languages manager `saveLanguage` controller method if it is detected that an update is in progress, this allows the language to be changed on the update screen. A similar exclusion already exists for installation.

I tried adding the `language=xxx` URL parameter to the updater URL and reloading the page instead of calling `saveLanguage`, but this only works for the initial updater page and since the language is not saved in the session then subsequent updater pages are not shown in the chosen language.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/LanguagesManager/Controller.php [hunk 1]
```diff
@@ -28,8 +29,13 @@ public function saveLanguage()
...
-        if (DbHelper::isInstalled()) {
+        // Check if is an update in progress
+        $updater = new PiwikCoreUpdater();
+        $updating = ($updater->getComponentUpdates() !== null);
+
+        // Prevent CSRF if Matomo is not installed yet or is updating
+        // (During install/update user can change language)
+        if (DbHelper::isInstalled() && !$updating) {
             $this->checkTokenInUrl();
         }
```
Comment (sgiehl) [inline plugins/LanguagesManager/Controller.php]: Actually I'm wondering if we shouldn't simply remove the whole token check from this method.
The `setLanguageForSession` called below does nothing else than setting a session cookie with the chosen language.
And we also have an nonce check, which makes it impossible to "trick" someone into changing his language.
In terms of security I can't see any requirement to check the token_auth 

@tsteur Can you see any security implications in removing the token check from this method?

Comment (tsteur) [reply plugins/LanguagesManager/Controller.php]: @sgiehl considering there is an nonce check I don't think the token check is needed. I haven't thought too much about it but it looks like both serve the same purpose and it's a duplicate.

Comment (sgiehl) [reply plugins/LanguagesManager/Controller.php]: @bx80 Let's remove the whole token check then.

Comment (bx80) [reply plugins/LanguagesManager/Controller.php]: Makes sense :+1:

**Review by sgiehl — APPROVED** [review]
Works as expected now

**Pull request merged**